devenv.nix

```diff
@@ -8,9 +14,30 @@
   languages.python.poetry.activate.enable = true;
   languages.python.version = "3.12";
 
+  git-hooks.hooks = {
+    # TODO: enable later
+    # ruff.enable = true;
+    # ruff-format.enable = true;
+    # typos.enable = true;
+    # check-toml.enable = true;
+    # markdownlint.enable = true;
+    yamllint.enable = true;
+    yamlfmt.enable = true;
+    yamlfmt.settings.lint-only = false;
+    nixfmt-rfc-style.enable = true;
+  };
+
   scripts.check-types.exec = ''
```

Comment:
this one errors too much as well currently, in order to run as part of CI. Will need to get the type errors down first.